src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/db/LastKnownAddressRepositoryIntegrationTest.kt

```diff
@@ -103,4 +105,37 @@ class LastKnownAddressRepositoryIntegrationTest : IntegrationTestBase() {
     createdByUserId,
     OffsetDateTime.now()
   )
+
+  @Test
+  @Transactional
+  fun `deleting a last known address that exists for recall`() {
+    val lastKnownAddressId = ::LastKnownAddressId.random()
+    recallRepository.save(recall, currentUserId)
+
+    lastKnownAddressRepository.saveAndFlush(lastKnownAddress(lastKnownAddressId, 1))
```

Comment:
Just 🤔  whether `saveAndFlush` allows for the retrieve via the recall to work in these tests too?
And whether we should move to `saveAndFlush` in place of `save` for all tests in this directory - with @Transactional on them all too, I'd assume.

Reply:
nope. It didn't make a difference. The only place it seems to is when we create recalls with documents.